evalml/pipelines/component_graph.py

```diff
@@ -231,6 +235,56 @@ def _compute_features(self, component_list, X, y=None, fit=False):
                 output_cache[component_name] = output
         return output_cache
 
+    def _get_feature_provenance(self, component_list, input_feature_names):
```

Comment:
Curious, why `component_list` as an argument? I noticed the only time we use this is in calling 
`self._feature_provenance = self._get_feature_provenance(self.compute_order, X.columns)`; would it make sense to just remove the arguments and internally use the component graph's compute_order? Or is this for flexibility? Are there use cases for this? (Maybe computing not the whole graph but some path in it? 🤔 )

Reply:
Great point! This is an oversight on my part. I think I was refactoring some code and thought I could make it a static method but that's not really reasonable. 

I'm deleting the component_list parameter now!